### PR TITLE
Ports making SSrunechat be a timer subsystem

### PR DESCRIPTION
**Pull request by ike709**
## About The Pull Request

Ports https://github.com/tgstation/tgstation/pull/64366

Includes some timer code we're missing that's in another open PR I have that hasn't been merged yet.

## Changelog
:cl: Bizzonium, ported by ike709
refactor: Made SSrunechat be a subsystem of SStimer
/:cl:

### CODE REVIEW

**Comment by ike709** [comment]
Probably confers misc. improvements, because SSrunechat was using old SStimer code from before some important fixes.

File: code/datums/chatmessage.dm [hunk 1]
```diff
@@ -544,8 +551,8 @@ GLOBAL_LIST_INIT(job_colors_pastel, list(
 	)
 
 	// Register with the runechat SS to handle EOL and destruction
-	scheduled_destruction = world.time + BALLOON_TEXT_TOTAL_LIFETIME(duration_mult)
-	enter_subsystem()
+	var/duration = CHAT_MESSAGE_LIFESPAN - CHAT_MESSAGE_EOL_FADE
+	fadertimer = addtimer(CALLBACK(src, .proc/end_of_life), duration, TIMER_STOPPABLE|TIMER_DELETE_ME, SSrunechat)
```
Comment (Archanial) [inline code/datums/chatmessage.dm]: this isn't runechat message, this is balloon alert

Comment (Archanial) [reply code/datums/chatmessage.dm]:
```suggestion
	var/duration = BALLOON_TEXT_TOTAL_LIFETIME(duration_mult)
	fadertimer = addtimer(CALLBACK(src, .proc/end_of_life), duration, TIMER_STOPPABLE|TIMER_DELETE_ME, SSrunechat)
```

Comment (ike709) [inline code/datums/chatmessage.dm]:
```suggestion
	var/duration = BALLOON_TEXT_TOTAL_LIFETIME(duration_mult)
	fadertimer = addtimer(CALLBACK(src, .proc/end_of_life), duration, TIMER_STOPPABLE|TIMER_DELETE_ME, SSrunechat)
```

**Review by Archanial — APPROVED** [review]

**Pull request merged**